src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1104 @@
+{
+    // does device mode allow this attribute to be updated?
+    if (!mDelegate->IsSupportedMapChangeAllowed())
+    {
+        return false;
+    }
+
+    if (mDelegate->ClearSupportedMaps())
+    {
+        ClearSupportedLocations();
+        NotifySupportedMapsChanged();
+        return true;
+    }
+
+    return false;
+}
+
+//*************************************************************************
+// Selected Locations manipulators
+
+bool Instance::AddSelectedLocation(uint32_t & aSelectedLocation)
+{
+    uint32_t dummyIndex;
+
+    char locationStatusBuffer[kMaxSizeStatusText];
+    MutableCharSpan locationStatusText(locationStatusBuffer);
+
+    // check max# of list entries
+    if (mDelegate->GetNumberOfSelectedLocations() >= kMaxNumSelectedLocations)
+    {
+        ChipLogError(Zcl, "AddSelectedLocation %u - maximum number of entries", aSelectedLocation);
+        return false;
+    }
+
+    // each item in this list SHALL match the LocationID field of an entry on the SupportedLocations attribute's list
+    if (!IsSupportedLocation(aSelectedLocation))
+    {
+        ChipLogError(Zcl, "AddSelectedLocation %u - not a supported location", aSelectedLocation);
+        return false;
+    }
+
+    // each entry in this list SHALL have a unique value
+    if (mDelegate->IsSelectedLocation(aSelectedLocation))
+    {
+        ChipLogError(Zcl, "AddSelectedLocation %u - duplicated location", aSelectedLocation);
+        return false;
+    }
+
+    // Does device mode allow modification of selected locations?
+    if (!mDelegate->IsSetSelectedLocationsAllowed(locationStatusText))
+    {
+        //        ChipLogError(Zcl, "AddSelectedLocation %u - %s", aSelectedLocation, locationStatusText);
+        return false;
+    }
+
+    return mDelegate->AddSelectedLocation(aSelectedLocation, dummyIndex);
+}
+
+bool Instance::ClearSelectedLocations()
+{
+    if (mDelegate->ClearSelectedLocations())
+    {
+        NotifySelectedLocationsChanged();
+        return true;
+    }
+
+    return false;
+}
+
+//*************************************************************************
+// Current Location manipulators
+
+DataModel::Nullable<uint32_t> Instance::GetCurrentLocation()
+{
+    return mCurrentLocation;
+}
+
+bool Instance::SetCurrentLocation(const DataModel::Nullable<uint32_t> & aCurrentLocation)
+{
+    // If not null, the value of this attribute SHALL match the LocationID field of an entry on the SupportedLocations attribute's
+    // list.
+    if ((!aCurrentLocation.IsNull()) && (!IsSupportedLocation(aCurrentLocation.Value())))
+    {
+        ChipLogError(Zcl, "SetCurrentLocation %u - location is not supported", aCurrentLocation.Value());
+        return false;
+    }
+
+    mCurrentLocation = aCurrentLocation;
+    NotifyCurrentLocationChanged();
+
+    // EstimatedEndTime SHALL be null if the CurrentLocation attribute is null.
+    if (mCurrentLocation.IsNull())
+    {
+        SetEstimatedEndTime(DataModel::Nullable<uint32_t>());
```

Comment:
```suggestion
        SetEstimatedEndTime(DataModel::NullNullable);
```

Reply:
This does not compile.

Reply:
fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/6cf2ea7077477afd319889a978cae71d5dc0429b).